File: ui/src/types/itinerary.ts
export interface ItineraryItem {
  id?: string;
  dayNumber: number;
  productId: string;
  quantity: number;
  notes?: string;
}

export interface Itinerary {
  id: string;
  leadCustomerId?: string | null;
  leadCustomerName?: string | null;
  startDate: string;
  duration: number;
  createdAt: string;
  items: ItineraryItem[];
}

export interface CreateItineraryRequest {
  startDate: string;
  duration: number;
  items: ItineraryItem[];
}

File: src/AI.Forged.TourOps.Domain/Entities/Itinerary.cs
namespace AI.Forged.TourOps.Domain.Entities;

public class Itinerary
{
    public Guid Id { get; set; }
    public DateOnly StartDate { get; set; }
    public int Duration { get; set; }
    public DateTime CreatedAt { get; set; }

    public ICollection<ItineraryItem> Items { get; set; } = new List<ItineraryItem>();
    public ICollection<Quote> Quotes { get; set; } = new List<Quote>();
}
